posthog/api/capture.py

```diff
@@ -404,9 +413,13 @@ def capture_internal(event, distinct_id, ip, site_url, now, sent_at, team_id, ev
     # overriding this to deal with hot partitions in specific cases.
     # Setting the partition key to None means using random partitioning.
     kafka_partition_key = None
-    candidate_partition_key = f"{team_id}:{distinct_id}"
 
-    if candidate_partition_key not in settings.EVENT_PARTITION_KEYS_TO_OVERRIDE:
-        kafka_partition_key = hashlib.sha256(candidate_partition_key.encode()).hexdigest()
+    if event["event"] in ("$snapshot", "$performance_event"):
+        kafka_partition_key = event["properties"]["$session_id"]
```

Comment:
What downstream assumptions would we break if we removed that locality and did simple round-robin instead?
If we keep it, we'll eventually want to introduce a sibling to `EVENT_PARTITION_KEYS_TO_OVERRIDE`.